src/Logging/LoggingTrait.php

```diff
@@ -0,0 +1,166 @@
+
+        // There was an error stringifying the event, return to not break execution
+        if ($stringifiedEvent === false) {
+            return;
+        }
+
+        $this->logger->debug($stringifiedEvent);
+    }
+
+    /**
+     * @param LogEvent $event
+     */
+    private function logResponse(LogEvent $event): void
+    {
+        $debugEvent = [
+            'timestamp' => $event->timestamp,
+            'severity' => strtoupper(LogLevel::DEBUG),
+            'clientId' => $event->clientId,
+            'requestId' => $event->requestId ?? null,
+            'jsonPayload' => [
+                'response.headers' => $event->headers,
+                'response.payload' => $event->payload,
+                'latency' => $event->latency,
+            ]
+        ];
+
+        // Remove null values
+        $debugEvent = array_filter($debugEvent, fn ($value) => !is_null($value));
+        $debugEvent['jsonPayload'] = array_filter(
+            $debugEvent['jsonPayload'],
+            fn ($value) => !is_null($value)
+        );
+
+        $stringifiedEvent = json_encode($debugEvent);
+
+        // There was an error stringifying the event, return to not break execution
+        if ($stringifiedEvent === false) {
+            return;
+        }
+
+        $this->logger->debug($stringifiedEvent);
+
+        if ($event->status) {
+            $infoEvent = [
+                'timestamp' => $event->timestamp,
+                'severity' => LogLevel::INFO,
+                'clientId' => $event->clientId,
+                'requestId' => $event->requestId ?? null,
+                'jsonPayload' => [
+                    'response.status' => $event->status
+                ]
+            ];
+
+            // Remove null values
+            $infoEvent = array_filter($infoEvent, fn ($value) => !is_null($value));
+
+            $stringifiedEvent = json_encode($infoEvent);
+
+            // There was an error stringifying the event, return to not break execution
+            if ($stringifiedEvent === false) {
+                return;
+            }
+
+            $this->logger->info($stringifiedEvent);
+        }
+    }
+
+    /**
+     * @param LogEvent $event
+     */
+    private function logStatus(LogEvent $event): void
+    {
+        $infoEvent = [
+            'timestamp' => $event->timestamp,
+            'severity' => LogLevel::INFO,
+            'clientId' => $event->clientId,
+            'requestId' => $event->requestId ?? null,
+            'jsonPayload' => [
+                'response.status' => $event->status
+            ]
+        ];
+
+        $infoEvent = array_filter($infoEvent, fn ($value) => !is_null($value));
+        $infoEvent['jsonPayload'] = array_filter(
+            $infoEvent['jsonPayload'],
+            fn ($value) => !is_null($value)
+        );
+
+        $this->logger->info((string) json_encode($infoEvent));
+    }
+
+    /**
+     * @param array<mixed> $headers
+     * @return null|array<mixed, mixed>
```

Comment:
we can make this a little smarter
```suggestion
     * @return null|array<string, string|false>
```